Support for DNS providers

eirslett opened the pull request:
This means you will get actual DNS records for your mi cluster, which makes it much easier to work with.

I have already tested this with SoftLayer and dnsimple. It would be great if somebody could test it with the other clouds as well!

eirslett commented on terraform/dnsimple/dns/main.tf:
```diff
@@ -0,0 +1,33 @@
+  count = "${var.worker_count}"
+  domain = "${var.domain}"
+  name = "*.service.${var.short_name}"
+  value = "${element(split(\",\", var.control_ips), count.index)}"
```
oops... this should be var.worker_ips. I'll fix it soon!

eirslett replied:
ok, this is fixed now!

ryane commented on terraform/gce/gce.tf:
```diff
@@ -130,3 +130,11 @@ resource "google_compute_instance" "mi-worker-nodes" {
+
+output "control_ips" {
+  value = "${join(\",\", google_compute_instance.mi-control-nodes.*.access_config.nat_ip)}"
+}
```
the gce outputs are not quite right. they should be something like:

```
google_compute_instance.mi-control-nodes.*.network_interface.0.access_config.0.nat_ip
google_compute_instance.mi-worker-nodes.*.network_interface.0.access_config.0.nat_ip
```

eirslett replied:
This should be fixed now. It's strange that it didn't work though, this is what the Terraform docs say: https://terraform.io/docs/providers/google/r/compute_instance.html#access_config

ryane commented:
so far, aws and digitalocean are working. the gce outputs are not correct (see inline comment). I'll test openstack next.

@eirslett would you be up for writing some docs explaining what the dnsimple integration does (what records it creates, etc.) and how to configure dnsimple credentials?

ryane commented:
openstack +1

eirslett commented:
Ok, I fixed the GCE stuff and added some documentation! I also changed to format of the DNS records for the load balancers. More info can be found in the commit messages.
Since we use SoftLayer, I can confirm this PR works there as well!

ryane commented:
Merged. @eirslett thanks for the nice addition
